lib/discourse_solved/first_accepted_post_solution_validator.rb

```diff
@@ -5,15 +5,15 @@ class FirstAcceptedPostSolutionValidator
     def self.check(post, trust_level:)
       return false if post.archetype != Archetype.default
       return false if !post&.user&.human?
-      return true if trust_level == "any"
 
-      return false if TrustLevel.compare(post&.user&.trust_level, trust_level.to_i)
-
-      if !UserAction.where(user_id: post&.user_id, action_type: UserAction::SOLVED).exists?
-        return true
+      if trust_level != "any"
+        return false if TrustLevel.compare(post&.user&.trust_level, trust_level.to_i)
       end
```

Comment:
If we don't plan to add more logic here, we can cut down the number of `if`s by combining the 2 conditionals (and possibly reduce indentation if the linter doesn't complain):

```suggestion
        return false if trust_level != "any" && TrustLevel.compare(post&.user&.trust_level, trust_level.to_i)
```

Reply:
🤔 

```
-      if trust_level != "any"
-        return false if TrustLevel.compare(post&.user&.trust_level, trust_level.to_i)
+      if trust_level != "any" && TrustLevel.compare(post&.user&.trust_level, trust_level.to_i)
+        return false
       end
```